### PR TITLE
feat(Forms): add `validator` support to Iterate.Array

### PR DESCRIPTION
**Pull request by tujoworker**
Quick example:

```tsx
<Form.Handler>
  <Iterate.Array
    path="/items"
    validator={(arrayValue) => {
      if (!(arrayValue?.length > 0)) {
        return new Error('You need at least one item')
      }
    }}
  >
    <Field.String itemPath="/" />
  </Iterate.Array>
  <Iterate.PushButton path="/items" pushValue="baz" />
  <Form.SubmitButton />
</Form.Handler>
```

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/FieldBlock/FieldBlock.tsx [hunk 1]
```diff
@@ -393,7 +393,7 @@ function FieldBlock(props: Props) {
     return null
   }
 
-  if (fieldState && !label) {
+  if (fieldState && !label && label !== false) {
     warn('You have to provide a label to use show an indicator.')
```
[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/FieldBlock/FieldBlock.tsx]: True! I revert this change and push it to a new [PR](https://github.com/dnbexperience/eufemia/pull/3927).

**Review by langz — APPROVED** [review]
Code looks good to me :) 
I've checked out this branch locally, and ran the Iterate stories to experience/test this myself, nothing in particular to report, other than it works 🥇 


Should we add the story to the docs as an demo/example, so that users more easily can see what this feature is, and how it's used? Or perhaps it's enough with the updated properties docs as it is.

**Comment by tujoworker** [comment]
> Should we add the story to the docs as an demo/example, so that users more easily can see what this feature is, and how it's used?

If you come over a meaningful use case? Maybe the minimum example is good enougth?

**Comment by langz** [comment]
> > Should we add the story to the docs as an demo/example, so that users more easily can see what this feature is, and how it's used?
> 
> If you come over a meaningful use case? Maybe the minimum example is good enougth?

Yeah, I think so 👍 
Can't come up with anything better on the fly 🪰

**Pull request merged**